### DISCUSSION TITLE
Microcontroller with 12-bit ADC and 10-bit DAC for Micropython

### DISCUSSION THREAD

**Discussion by WolleKette** [micropython/micropython#11093] [Hardware & Peripherals] [answered]
I am looking for a microcontroller with 12-bit ADC and 10-bit DAC, programmable in Micropython. Otherwise, there are no special requirements. Can someone recommend me a module?

**Comment by robert-hh** [comment, marked as answer]
e.g. Pyboard, SAMD51 modules. The SAMD51 has 12 bit ADC and 12 bit DAC. The SAM21 has 12 bit ADC and 10 bit DAC.

**Comment by bixb922** [comment]
ESP32 has 12 bit ADC but only 8 bit DAC (https://github.com/micropython/micropython/blob/a9bbf7083ef6b79cf80bdbf34984d847a6c4aae9/ports/esp32/machine_dac.c), so that does not qualify. The RP2040 has 12 bit ADC but no DAC. Obviously one way is to use an external module, but that increases component count...
